clients/go/pkg/auditopt/config.go

```diff
@@ -269,6 +246,15 @@ func labelsFromConfig(cfg *api.Config) audit.Option {
 	return audit.WithMutator(&lp)
 }
 
+func justificationFromConfig(ctx context.Context, cfg *api.Config) (audit.Option, error) {
```

Comment:
1. I don't like the idea of creating a new justification client each call.
2. This client is never closed, which could leak fd over time.
3. The threading of `ctx` from parsing the config seems incorrect here.

Reply:
RE 1: This function is supposed to be called once (e.g. in main.go) when creating an audit client. E.g. https://github.com/abcxyz/lumberjack/blob/main/clients/go/test/shell/main.go#L123. In addition, it's possible to not use `auditopt` package to load options from a config file, in which case people can provide a reusable justification client as an option.

RE 2: Not sure I follow. The justification client doesn't have a "close" func. The audit client should be stopped by the caller.

RE 3: How come? The `ctx` is not per-request context.

Reply:
1. Okay - that wasn't immediately clear from the PR. Maybe we could add some code comments to clarify the ctx scope.

2. I thought this was a grpc client library (which needs to be closed). I see it's our client library now which does not require closing.

3. See 1

Reply:
Thanks for the callout!

We have some existing doc explaining the usage: https://github.com/abcxyz/lumberjack/blob/main/docs/clients.md#go
But I can see room for improvements and package level examples as godoc. Created an issue: https://github.com/abcxyz/lumberjack/issues/289

Reply:
`Okay - that wasn't immediately clear from the PR. Maybe we could add some code comments to clarify the ctx scope`, similar usage is also used when we create a logging client https://github.com/abcxyz/lumberjack/blob/f03fda0c297d43a15ce2d367c44b7b1a7c233dfe/clients/go/pkg/auditopt/config.go#L250, instead of adding comments, it will be better to add docs in #289